bsi_zoo/benchmark.py

```diff
@@ -162,16 +214,18 @@ def run(self, nruns=2):
         }
 
         estimators = [
-            (iterative_L1, data_args_I, {"alpha": estimator_alphas}),
-            (iterative_L2, data_args_I, {"alpha": estimator_alphas}),
-            (iterative_sqrt, data_args_I, {"alpha": estimator_alphas}),
-            (iterative_L1_typeII, data_args_II, {"alpha": estimator_alphas}),
-            (iterative_L2_typeII, data_args_II, {"alpha": estimator_alphas}),
-            (gamma_map, data_args_II, {"alpha": estimator_alphas}),
+            # (iterative_L1, data_args_I, {"alpha": estimator_alphas}, {}),
+            # (iterative_L2, data_args_I, {"alpha": estimator_alphas}, {}),
+            # (iterative_sqrt, data_args_I, {"alpha": estimator_alphas}, {}),
+            # (iterative_L1_typeII, data_args_II, {"alpha": estimator_alphas}, {}),
+            # (iterative_L2_typeII, data_args_II, {"alpha": estimator_alphas}, {}),
+            (gamma_map, data_args_II, {"alpha": estimator_alphas},  {"update_mode": 1}),
+            (gamma_map, data_args_II, {"alpha": estimator_alphas},  {"update_mode": 2}),
+            (gamma_map, data_args_II, {"alpha": estimator_alphas},  {"update_mode": 3}),
         ]
 
         df_results = []
-        for estimator, data_args, estimator_args in estimators:
+        for estimator, data_args, estimator_args, estimator_extra_params in estimators:
             benchmark = Benchmark(
                 estimator,
```

Comment:
one way to do this would be to pass here an instance of Solver or SpatialCVSolver instead of the function like gamma_map. Just an idea